### ISSUE TITLE
[Accessibility] Screen reader does not announce Send failed message information

### ISSUE THREAD

**Issue by compulim** [microsoft/BotFramework-WebChat#4211]
> Copied from [Composer#64488](https://fuselabs.visualstudio.com/Composer/_workitems/edit/64488).

## Version

`4.15.1` (or `main`)

## Describe the bug

If screen reader users navigate on Live Chat Box screen and message get failed and message appear send failed and screen reader does not announce this message, then it would be difficult for users to know sent message get failed and need to try again to send the message.

## Steps to reproduce

1. Open Bot Framework Emulator installed on the system.
1. Bot Framework Emulator Application with Welcome screen opens.
1. Navigate on the welcome screen and select Open Bot button.
1. Open A bot Dialog opens, navigate on the dialog and enter inputs in edit fields and select Connect button.
1. Bot Gets connected and a new Tab "Live Chat "opens, Navigate on the tab.
1. Verify that screen reader announces send failed message or not.

## Expected behavior

Screen reader should be announcing Send failed message information under Live Chat Box.
So that users know send message get failed, it does not deliver to another person.

## Additional context

Related to [WCAG 2.1 1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html).

Some texts from designer:

```
Message failed to send. Go to message to resend.

Your message failed to send: .......... Retry.
```

[Bug]

**Comment by compulim** [comment]
Please work with #4322 together.

Action items:

- Create a new `useFailedActivityKeys(): [string]` to list all failing activities
   - Mind that `props.sendTimeout` and `props.sendTimeoutForAttachment` could change at any time
   - It is okay to say "`sendTimeout` change will affect next activity, but not any current activities". But it's better if we could make it affect current activities
   - This hook will trigger re-render when an activity failed to send
- Modify all existing code to use the new `useFailedActivityKeys` hook
- Verify when `postActivity()` return failing `Observable`, the send status should turn to failed immediately (without waiting for 20 seconds)
- In `<LiveRegionTranscript>`, when a message is added to the `useFailedActivityKeys`, queue it to live region
